intranet/apps/printing/views.py

```diff
@@ -245,84 +263,97 @@ def print_job(obj: PrintJob, do_print: bool = True):
-    with open(tmpfile_name, "wb+") as dest:
-        for chunk in fileobj.chunks():
-            dest.write(chunk)
 
-    tmpfile_name = convert_file(tmpfile_name, filebase)
+    # The paths to files that we need to clean up at the end
+    # This needs to be a set because we may add duplicate entries to it
+    delete_filenames = set()
+    try:
+        tmpfile_fd, tmpfile_name = tempfile.mkstemp(prefix="ion_print_{}_{}".format(obj.user.username, filebase_escaped), text=False)
+        delete_filenames.add(tmpfile_name)
 
-    if not tmpfile_name:
-        msg = "Error converting file to PDF for printing"
+        # This implicitly closes tmpfile_fd when it's done writing
+        with open(tmpfile_fd) as dest:
+            for chunk in fileobj.chunks():
+                dest.write(chunk)
 
-        if filebase.endswith((".doc", ".docx")):
-            msg += (
-                "<br>Note: It looks like you are trying to print a Word document. Word documents don't always print correctly, so we recommend "
-                "that you convert to a PDF before printing."
-            )
+        final_filename = convert_file(tmpfile_name, filebase)
+        delete_filenames.add(final_filename)
 
-        raise Exception(msg)
+        if not final_filename:
+            msg = "Error converting file to PDF for printing"
 
-    if get_mimetype(tmpfile_name) == "text/plain":
-        with open(tmpfile_name, "r") as f:
-            num_chars = sum(len(line) for line in f)
-        num_pages = num_chars // (settings.PRINTING_PAGES_LIMIT * 50 * 72)
-    else:
-        num_pages = get_numpages(tmpfile_name)
-        if num_pages < 0:
-            raise Exception("Could not get number of pages in {}".format(filebase))
-
-    if re.search(r"\d\s+\d", obj.page_range) is not None:
-        # Make sure that when removing spaces in the page range we don't accidentally combine two numbers
-        raise InvalidInputPrintingError(
-            "You specified an invalid page range (please separate page numbers with 1) commas to print selected pages or 2) dashes to print a range)."
-        )
+            if filebase.endswith((".doc", ".docx")):
+                msg += (
+                    "<br>Note: It looks like you are trying to print a Word document. Word documents don't always print correctly, so we recommend "
+                    "that you convert to a PDF before printing."
+                )
 
-    obj.num_pages = num_pages
-    obj.page_range = "".join(obj.page_range.split())  # remove all spaces
-    obj.save()
+            raise Exception(msg)
 
-    range_count = check_page_range(obj.page_range, obj.num_pages)
+        if get_mimetype(final_filename) == "text/plain":
+            with open(final_filename, "r") as f:
+                num_chars = sum(len(line) for line in f)
+            num_pages = num_chars // (settings.PRINTING_PAGES_LIMIT * 50 * 72)
```

Comment:
This isn't related or blocking, but do you know the reasoning behind 50 and 72?

Reply:
I'm guessing 72 characters, 50 lines a page?

While I'm at it, I'll do some refactoring to make this 1) detect line wrapping and 2) just go off a single "lines per page" constant.